### PR TITLE
Pou 49 search bar filters minus dates

### PR DESCRIPTION
**Pull request by MingLiang-PALOIT**
[POU 49](https://cadtrust.atlassian.net/browse/POU-49)

# Why

Search filters on the home page

# What

- Search bar
- Multi-select filters with auto complete
- Other small UI changes

### CODE REVIEW

File: src/components/molecules/AutoCompleteCheckbox/AutoCompleteCheckbox.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            <Flex width={`100%`} marginTop={`4px`} alignItems={`center`} padding={`8px 0.75rem`}>
+              <Checkbox
+                isIndeterminate={isIndeterminate}
+                isChecked={allChecked}
+                onChange={() => {
+                  if (isIndeterminate || selectedValues.length === 0) {
+                    setSelectedValues([...options])
+                  } else {
+                    setSelectedValues([])
+                  }
```
Comment (morphet81) [inline src/components/molecules/AutoCompleteCheckbox/AutoCompleteCheckbox.tsx:68]: Never put events handler inline. Always declare functions.

File: src/components/molecules/AutoCompleteCheckbox/AutoCompleteCheckbox.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          <VStack divider={<StackDivider height={`1px`} borderColor={`#B8BEC0`} />} spacing={0}>
+            <Flex width={`100%`} alignItems={`center`} padding={`8px`}>
+              <SearchIcon color={`lightGray.700`} paddingY={`auto`} marginRight={`8px`} />
+              <Input
+                placeholder={`Search`}
+                variant={`autoCompleteCheckboxTextInput`}
+                value={searchInput}
+                onChange={(event) => {
+                  setSearchInput(event.target.value)
+                }}
```
Comment (morphet81) [inline src/components/molecules/AutoCompleteCheckbox/AutoCompleteCheckbox.tsx:56]: Never put events handler inline. Always declare functions.

File: src/components/molecules/AutoCompleteCheckbox/AutoCompleteCheckbox.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  return (
+    <Popover
+      gutter={0}
+      isLazy
+      placement="bottom-start"
+      matchWidth
+      onClose={() => {
+        setSearchInput(``)
+        setSelectedValues([...selectedFilters])
+      }}
```
Comment (morphet81) [inline src/components/molecules/AutoCompleteCheckbox/AutoCompleteCheckbox.tsx:32]: Never put events handler inline. Always declare functions.

File: src/overmind/searchFilters/effects.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+          const governanceResponseData = body.data as GovernanceResponseData
+          result = { data: governanceResponseData }
+        } else {
+          result = { error: { code: body.status.toString(), message: body.statusText } }
+        }
+        resolve(result)
+      })
+      .catch(() => {
+        result = { error: { code: `400`, message: `could not fetch data` } }
+        resolve(result)
```
Comment (morphet81) [inline src/overmind/searchFilters/effects.ts:24]: Are you sure you want to resolve on error instead of reject?

Comment (MingLiang-PALOIT) [reply src/overmind/searchFilters/effects.ts:23]: This has been the pattern that was established by Mederic previously. The EffectResponse contains both a generic data perimeter and an ErrorResponse.

File: src/components/molecules/AutoCompleteCheckbox/AutoCompleteCheckbox.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    }
+  }
+
+  const checkboxOnChange = (checked: boolean, value: string) => {
+    if (checked) {
+      setSelectedValues([...selectedValues, value])
+    } else {
+      setSelectedValues((prevState) => prevState.filter((selectedValue) => selectedValue !== value))
+    }
+  }
```
Comment (morphet81) [inline src/components/molecules/AutoCompleteCheckbox/AutoCompleteCheckbox.tsx:49]: event handler functions should start with `handle`

File: src/overmind/searchFilters/effects.ts (hunk 4 above)
Comment (morphet81) [reply src/overmind/searchFilters/effects.ts:23]: @crimson-med  and @MingLiang-PALOIT this is not the way this kind of pattern should be written. You should rely on promise lifecyle:

1. before starting the job, init an empty response object
2. within `then` set the data
3. within `catch` set the error
4. within `finally` resolve your promise

**Review by morphet81 — CHANGES_REQUESTED** [review]

File: src/components/molecules/AutoCompleteCheckbox/AutoCompleteCheckbox.tsx (hunk 5 above)
Comment (MingLiang-PALOIT) [reply src/components/molecules/AutoCompleteCheckbox/AutoCompleteCheckbox.tsx]: have refactored the functions to start with `handle`

**Review by morphet81 — APPROVED** [review]

**Pull request merged**